src/gwt/src/org/rstudio/studio/client/RStudio.java

```diff
@@ -157,9 +159,13 @@ public void execute()
       };
    }
    
+   private static final native void maybeSetWindowName(String name)
+   /*-{
+      $wnd.name = $wnd.name || name;
```

Comment:
It seems like this won't set the window name if it already has one (`"abc" || "123"` evaluates to `"abc"`). Is that intentional?

Reply:
Indeed it is -- for example, RStudio satellite windows will run through this code path, but will be opened with an explicit satellite name and I wanted to avoid overwriting that.

Reply:
Okay, great!